File: TP2/TP2Q10/StarWarsSelecao.c
/*
Nome: Larissa Domingues Gomes
Turno: Manha
Numero de Matricula: 650525
Professor: Felipe Cunha 
Questao:Classe Personagens Star Wars C Pesquisa binaria
*/

#include <time.h>
#include "ListaPersonagem.h"


bool IsFim(char p[])
{
   return (strlen(p) == 3 && p[0] == 'F' && p[1] == 'I' && p[2] == 'M');
}
 
int main()
{
   clock_t t1 = clock();
   int comp = 0;
   int sw = 0;
   FILE* arq = fopen("matrícula_selecaoRecursiva.txt", "wt");
   char leitura[_MAX_];
   ListaPersonagem l;
   ConstrutorLista(&l, 90);
   scanf("%s", leitura);
   getchar();//limpar buffer
   //Repeticao para fazer o set de dados do array de personagens ate o fim
   while(!IsFim(leitura)) {
      inserirFim(leitura, &l);
      scanf("%s", leitura);
      getchar();//limpar buffer
   }           
   scanf(" %[^\n]s", leitura);
  //printf("%s\n",leitura);
   
   selecao(&l,0, 0,1,&comp,&sw);
   printLista(&l);                
   freeLista(&l);
   clock_t t2 = clock();
   fprintf(arq, "650525\t%i\t%i\t%lf",comp,sw, ((t2-t1)/1000.0));
   fclose(arq);
}

File: TP2/TP2Q09/StarWarsSelecao.java
/*
Nome: Larissa Domingues Gomes
Turno: Manha
Numero de Matricula: 650525
Professor: Felipe Cunha 
Questao:Classe Personagens Star Wars Java Ordenacao por Selecao
*/

import java.io.FileWriter;
import java.io.PrintWriter;
import java.util.Date;


public class StarWarsSelecao{
   public static void main(String[] args)throws Exception
   {
      long inicio = new Date().getTime();
      ListaPersonagem p = new ListaPersonagem(90); 
      FileWriter arquivo = new FileWriter("matrícula_sequencial.txt");
      PrintWriter gravar = new PrintWriter(arquivo);
      String arq = MyIO.readLine();

      //Repeticao para fazer o set de dados do array de personagens ate arq = fim
      while(!arq.equals("FIM")) {
         p.inserirFim(arq);
         arq = MyIO.readLine();
      }  
      p.selecao();
      p.printLista();
      long fim = new Date().getTime();
      gravar.println("650525\t"+p.getNumComparacoes()+"\t"+p.getNumMovimentacoes()+"\t"+(fim-inicio)/1000.0);
      //gravar no arquivo numero de comparacoes e tempo de execucao
      arquivo.close();
   }
}
